pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -332,6 +332,20 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(
 			continue
 		}
 
+		// If we delete the pod and the configmap used by the pod when kubelet is stoped, after kubelet started, the configmap volume will mount failed forever.
+		// so we should not add the configmap volume to desiredStateOfWorld when pod's DeletionTimestamp is not empty.
+		if pod.DeletionTimestamp != nil && podVolume.VolumeSource.ConfigMap != nil {
```

Comment:
Special-casing secret/configmap volumes here seems strange. What about:
* projected volumes sourced from secret/configmap objects
* projected volumes sourced from service accounts
* PVC volumes whose PVC object has been removed

Does this also assume termination will be short and it is ok to stop updating secret/configmap volumes during that time? Is that a good assumption (containers can take significant time to terminate)?

Is existing secret/configmap content left as-is if we omit these or is it removed? If it is removed, can that break a container while it is terminating?